File: 17solve/17-155.pas
// Автор В.Н. Шубинкин
##
var a := ReadLines('17-1.txt').Select(t -> t.ToInteger).ToArray;
var count := 0;
var maxEl := -20000;
for var i := 1 to a.High - 1 do
begin
  if (a[i] < a[i - 1]) and (a[i] < a[i + 1]) then
  begin
    count += 1;
    maxEl := max(maxEl, a[i])
  end;
end;
print(count, maxEl)

File: 17solve/17-155.py
# Автор В.Н. Шубинкин

with open('17-1.txt') as f:
    a = [int(x) for x in f.readlines()]

count = 0
mx = -20000
for i in range(1, len(a) - 1):
    if a[i - 1] > a[i] < a[i + 1]:
        count += 1
        mx = max(mx, a[i])
print(count, mx)
